### PR TITLE
Openai embeddings 3

### PR DESCRIPTION
**Pull request by yisding**
Also supporting the new GPT-4 preview model

### CODE REVIEW

**Review by seldo — APPROVED** [review]
LGTM

File: packages/core/src/embeddings/OpenAIEmbedding.ts [hunk 1]
```diff
@@ -30,7 +44,8 @@ export class OpenAIEmbedding extends BaseEmbedding {
   constructor(init?: Partial<OpenAIEmbedding> & { azure?: AzureOpenAIConfig }) {
     super();
 
-    this.model = init?.model ?? OpenAIEmbeddingModelType.TEXT_EMBED_ADA_002;
+    this.model = init?.model ?? "text-embedding-ada-002";
+    this.dimensions = init?.dimensions;
```
Comment (EmanuelCampos) [inline packages/core/src/embeddings/OpenAIEmbedding.ts:48]: backward compatibility’s would be good to have a default dimensions as well

Comment (yisding) [reply packages/core/src/embeddings/OpenAIEmbedding.ts:48]: Good point. So the way this works is the dimensions is undefined if the user doesn't specify so it automatically uses the default dimensions of the model (by not sending the parameter) but agreed this is not obvious just from the code. Will add some comments to make this clearer.

**Review by himself65 — APPROVED** [review]
lgtm

File: packages/core/src/embeddings/OpenAIEmbedding.ts (hunk 1 above)
Comment (yisding) [reply packages/core/src/embeddings/OpenAIEmbedding.ts]: https://github.com/run-llama/LlamaIndexTS/pull/445/commits/f27fd93b881c6ecbdc257d1286724f0faf8ee0c4 Really appreciate the feedback Emanuel!

**Pull request merged**